lldb/tools/lldb-dap/DAP.cpp

```diff
@@ -1437,7 +1437,12 @@ void DAP::EventThread() {
           const bool remove_module =
               event_mask & lldb::SBTarget::eBroadcastBitModulesUnloaded;
 
-          std::lock_guard<std::mutex> guard(modules_mutex);
+          // NOTE: Both mutexes must be acquired (API mutex first, then modules
+          // mutex) to prevent deadlock when handling `modules_request`, which
+          // also requires both locks.
+          lldb::SBMutex api_mutex = GetAPIMutex();
+          std::lock_guard api_guard(api_mutex);
+          std::lock_guard modules_guard(modules_mutex);
```

Comment:
Can we use scoped_lock (C++17)?
```suggestion
          std::scoped_lock lock(api_mutex, modules_mutex);
```

Reply:
updated to use `std::scoped_lock`